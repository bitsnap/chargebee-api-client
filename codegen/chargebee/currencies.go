package chargebee

import (
	"github.com/bitsnap/chargebee-api-client/codegen/client"
)

// GenerateListCurrencies generates chargebee client code to fetch all currencies
//
// API: https://{site}.chargebee.com/api/v2/currencies
//
// Documentation: https://apidocs.chargebee.com/docs/api/currencies?lang=curl#list_currencies
func GenerateListCurrencies() string {
	return client.GenerateList("Currencies", "Currency", "chargebee.com/api/v2/currencies")
}

// GenerateRetrieveCurrency generates chargebee client code to retrieve specific currency
//
// API: https://{site}.chargebee.com/api/v2/currencies/{currency-id}
//
// Documentation: https://apidocs.chargebee.com/docs/api/currencies?lang=curl#retrieve_an_currency
func GenerateRetrieveCurrency() string {
	return client.GenerateRetrieve("Currency", "chargebee.com/api/v2/currencies")
}

// GenerateCurrenciesModel generates chargebee currency domain model
//
// Documentation: https://apidocs.chargebee.com/docs/api/currencies?lang=curl#currency_attributes
func GenerateCurrenciesModel() string {
	return client.GenerateModel("Currency", "Currency attributes", "https://apidocs.chargebee.com/docs/api/currencies")
}
