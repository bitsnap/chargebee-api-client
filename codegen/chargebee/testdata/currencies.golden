# Generated by goldga. DO NOT EDIT.
[snapshots]
"Chargebee currencies code generation when generating Chargebee API client should be able to generate currencies models" = '''
(string) (len=315) "type Currency struct {\n    Id string `json:\"id\"`\n    Enabled bool `json:\"enabled\" validate:\"required\"`\n    ForexType enums.ForexTypeEnum `json:\"forex_type\"`\n    CurrencyCode string `json:\"currency_code\"`\n    IsBaseCurrency bool `json:\"is_base_currency\"`\n    ManualExchangeRate string `json:\"manual_exchange_rate\"`\n}"
'''
